### ISSUE TITLE
Compiling a project with a single dependency on MinTTY fails in lts-18.17

### ISSUE THREAD

**Issue by theGhostJW** [commercialhaskell/lts-haskell#329]
**Resolver**
```bash
resolver:
  url: https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/lts/18/17.yaml
```

**Log**
```bash
PS C:\MinttyProblem> stack build
WARNING: Ignoring mintty's bounds on Win32 (>=2.13.1); using Win32-2.6.2.1.
Reason: trusting snapshot over cabal file dependency information.
mintty    > configure
mintty    > Configuring mintty-0.1.3...
mintty    > build
mintty    > Preprocessing library for mintty-0.1.3..
mintty    > Building library for mintty-0.1.3..
mintty    > [1 of 1] Compiling System.Console.MinTTY
mintty    >
mintty    > src\System\Console\MinTTY.hs:31:1: error:
mintty    >     Could not find module `System.Console.MinTTY.Win32'
mintty    >     Use -v (or `:set -v` in ghci) to see a list of the files searched for.
mintty    >    |
mintty    > 31 | import qualified System.Console.MinTTY.Win32 as Win32 (isMinTTY, isMinTTYHandle)
mintty    >    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Demo Project**
https://github.com/theGhostJW/MinttyProblem.git

**Related Chatter / Work Around**
- https://www.reddit.com/r/haskell/comments/qksh4o/monthly_hask_anything_november_2021/hlee653/?context=3
- https://www.reddit.com/r/haskell/comments/qwsvg4/minttywin32/

Please let me know if this not the right place to be logging this.
thanks

**Comment by mpilgrem** [comment]
@DanBurton, @juhp I think this is unresolved in `lts-18.20`. (I only came across this repository after searching for, and then raising, the issue in the `stackage` repository.)

**Comment by juhp** [comment]
I agree we should fix this before lts19 at least

**Comment by alexeyzab** [comment]
Should be resolved in `lts-18.21`, going to close this for now.
